pkg/services/ngalert/state/manager.go

```diff
@@ -161,6 +164,17 @@ func (st *Manager) ProcessEvalResults(ctx context.Context, alertRule *ngModels.A
 	return states
 }
 
+func (st *Manager) takeScreenshot(ctx context.Context, alertRule *ngModels.AlertRule, state *State) error {
+	if state.Screenshot.ImageOnDiskPath == "" {
```

Comment:
Can you put more explanation here why we redo the screenshot if it is not stored on disk but (presumably) is available by URL?

Reply:
The thinking here is that for a screenshot to be available via an URL it must also be on disk. Is it possible for Grafana to have a screenshot via a URL and not on disk? I suppose if someone goes into the fs and deletes the files?